modules/openapi-generator/src/main/java/org/openapitools/codegen/DefaultGenerator.java

```diff
@@ -194,14 +194,8 @@ private void configureGeneratorProperties() {
 
         URL url = URLPathUtils.getServerURL(openAPI);
         contextPath = config.escapeText(url.getPath());
-        basePathWithoutHost = contextPath; // for backward compatibility
-        basePath = config.escapeText(URLPathUtils.getHost(openAPI));
-        if ("/".equals(basePath.substring(basePath.length() - 1))) {
-            // remove trailing "/"
-            // https://host.example.com/ => https://host.example.com
-            basePath = basePath.substring(0, basePath.length() - 1);
-        }
-
+        basePathWithoutHost = contextPath.replaceAll("/$", ""); // for backward compatibility
```

Comment:
@bjgill for `contextPath`, I think you will need to do the same by removing the trailing "/". I remember a few generators are still using `contextPath` in the templates